src/lib/storage/storage_manager.cpp

```diff
@@ -5,33 +5,38 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  // Will refer to the same instance on all method invocations and can therefore be used as a singleton.
+  static StorageManager INSTANCE;
+  return INSTANCE;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  _table_names.emplace_back(name);
+  _tables.emplace_back(table);
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(has_table(name), "Cannot drop non-existing table.");
+  auto index = find(_table_names.begin(), _table_names.end(), name);
+
+  _table_names.erase(index);
+  _tables.erase(_tables.begin() + (index - _table_names.begin()));
 }
 
 std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(has_table(name), "Cannot find table.");
```

Comment:
You should think about adapting either Assert (appear in release builds) or DebugAssert (do not appear in release builds) in all cases to be concise amongst the whole codebase.

```suggestion
  DebugAssert(has_table(name), "Cannot find table.");
```

Reply:
I'm not sure this is true in general. After all, there is a reason why `Assert` and `DebugAssert` are two different things, so we should probably think about which one to use for each case separately.

Reply:
If you have good reasoning for this kind of `Assert`, than it should be fine :)